### DISCUSSION TITLE
Listen for PongWebSocketFrame in class annotated @ServerWebSocket

### DISCUSSION THREAD

**Discussion by bossman0072** [micronaut-projects/micronaut-core#5246] [Q&A]
My websocket server is sending PingWebSocketFrame and then receives PongWebSocketFrame. How could I listen for that frame?
Source of AbstractNettyWebSocketHandler has code like this:
`if (msg instanceof PongWebSocketFrame) {
                    return;
 }`
As I understand there is no way for handling PongWebSocketFrame. Am I right?

**Comment by graemerocher** [comment]
Looks like a limitation, what is the use case for listening for this frame?

**Reply by bossman0072** [reply]
Check if client is still connected. There is no way to change client code and add support for custom ping message.
